### ISSUE TITLE
visibility-modifier-order false positive in case of payable argument and custom modifier

### ISSUE THREAD

**Issue by ZumZoom** [protofire/solhint#249]
```solidity
contract C {
    modifier m() {
        _;
    }

    function foo(address payable addr) public payable m {
        addr;
    }
}
```

This contract triggers `Visibility modifier must be first in list of modifiers` however visibility modifier is already first in list of modifiers. Issue is reproduced only with `address payable` argument, `payable` modifier and a custom modifier.
